platform/server/src/main/java/org/motechproject/server/bootstrap/BootstrapController.java

```diff
@@ -168,6 +182,27 @@ public ModelAndView submitForm(@ModelAttribute(BOOTSTRAP_CONFIG) @Valid Bootstra
         return response;
     }
 
+    @RequestMapping(value = "/verifyAmq", method = RequestMethod.POST)
+    @ResponseBody
+    public Map<String, ?> verifyAmqConnection(@ModelAttribute(BOOTSTRAP_CONFIG) @Valid BootstrapConfigForm form,
+                                              BindingResult result, HttpServletRequest request) {
+        Map<String, Object> response = new HashMap<>();
+        if (result.hasErrors()) {
+            response.put(WARNINGS, Arrays.asList(getMessage("server.bootstrap.verify.amq.error", request)));
+            response.put(SUCCESS, false);
+        } else {
+            String queueUrl = form.getQueueUrl();
+            boolean reachable = messageBrokerPingService.pingBroker(queueUrl);
+            if (reachable) {
+                response.put(SUCCESS, reachable);
+            } else {
+                response.put(WARNINGS, Arrays.asList(String.format(getMessage("server.bootstrap.verify.amq.warning", request), queueUrl)));
```

Comment:
Instead of doing String.format, use placeholders in messages and use the messageSource with additional arguments